src/main/java/com/dawid/customers/controller/CustomerController.java

```diff
@@ -38,9 +38,17 @@ public List<Customer> findCustomerByAge(@PathVariable(value = "age") Integer age
         return customerRepository.findCustomersByAge(age);
     }
 
-    @GetMapping("/name/{firstname}")
-    public List<Customer> findCustomerByFirstName(@PathVariable(value = "firstname") String firstname) {
+    @GetMapping("/firstname")
+    public List<Customer> findCustomerByFirstName(String firstName) {
 
-        return customerRepository.findCustomersByFirstName(firstname);
+        return customerRepository.findCustomersByFirstName(firstName);
+    }
+    @DeleteMapping("/firstName")
+    public List<Customer> deleteCustomerByFirstName(String firstName){
+        return customerRepository.deleteCustomersByFirstName(firstName);
+    }
+    @PutMapping("/id/{id}")
+    public List<Customer> updateCustomerById(Long id){
```

Comment:
Shouldn't we return an updated Customer instance here as opposed to the List<Customer>